### DISCUSSION TITLE
How can I close the notification inbox using JavaScript code?

### DISCUSSION THREAD

**Discussion by 3v0k4** [magicbell/community#29] [Questions] [answered]
See title. Thanks! 🙏

**Comment by smeijer** [comment, marked as answer]
If you need to close the inbox programatically, you'll need to take over control over the `isOpen` prop. The approach depends a bit on what SDK you're using.

## [@magicbell/magicbell-react](https://github.com/magicbell-io/magicbell-js/tree/main/packages/react)

When using react, you'll need to control the `isOpen` prop. The exact implementation depends very much on your requirements, but here's the basic approach. 

```jsx
function MyApp() {
  const { showInbox } = useContext(AppContext);

  return (
    <MagicBell apiKey={apiKey} userEmail="person@example.com">
      {(props) => (
        <FloatingNotificationInbox {…props} isOpen={showInbox} />
      )}
    </MagicBell>
  );
}
```

## [@magicbell/embeddable](https://github.com/magicbell-io/magicbell-js/tree/main/packages/embeddable)

The embeddable currently doesn't expose a way to take over control. You could query the DOM to check the current state tho, and then simulate a click event to close it.

```jsx
const magicbell = document.querySelector('[data-magicbell-bell]');
const expanded = magicbell.closest('[aria-expanded]').getAttribute('aria-expanded') === 'true';
if (expanded) magicbell.click();
```
